lib/hooks/core/hook-http.js

```diff
@@ -44,7 +44,10 @@ function requestWrap(request) {
     var root = cls.getRootContext();
 
     if (!root) {
-      agent.logger.debug('Untraced http options:', options);
+      if (typeof options === 'string') {
+        options = url.parse(options);
```

Comment:
nevermind. There is a return following this statement, so there is no merge with the mainline path. This code is good as it is now.